engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorContextImpl.java

```diff
@@ -107,6 +111,12 @@ public HibernateValidatorContext clockProvider(ClockProvider clockProvider) {
 	}
 
 	@Override
+	public HibernateValidatorContext addValueExtractor(ValueExtractor<?> extractor) {
+		valueExtractors.add( extractor );
```

Comment:
Maybe we could change them to VED here and throw the error early if it's not a valid VE?